pennylane/operation.py

```diff
@@ -517,6 +515,20 @@ def __repr__(self):
             return "{}({}, wires={})".format(self.name, params, self.wires.tolist())
         return "{}(wires={})".format(self.name, self.wires.tolist())
 
+    @property
+    def num_params(self):
+        """Number of trainable parameters that this operator expects to be fed via the
+        dynamic `*params` argument.
+
+        By default, this property returns as many parameters as were used for the
+        operator creation. If the number of parameters for an operator subclass is fixed,
+        this property can be overwritten to return the fixed value.
+
+        Returns:
+            int: number of parameters
+        """
+        return self._num_params
```

Comment:
If this is a good way to do it we could implement `num_wires` in the same way and remove `AnyWires` etc!

Reply:
oh, good idea! Although, I wonder if there is logic somewhere that needs to know if an (uninstantiated) operation applies to any wires...

Reply:
Good to check first, indeed!